stages/org.osbuild.coreos.platform

```diff
@@ -0,0 +1,115 @@
+where each platform has its own provided disk image with slightly
+differing settings/behavior. This stage will perform the necessary
+configuration for the given platform. This configuration boils down to
+a few steps:
+1. Locate the source of platform specific information that is provided
+   in the CoreOS filesystem tree already (the platforms.json).
+2. Copy the platforms.json file into the /boot/ partition, which is
+   sometimes used by coreos-installer.
+3. Read the platforms.json to fetch and platform specific kernel
+   arguments or grub configuration to set. These arguments/config
+   are primarily console settings.
+4. Apply any platform specific kernel arguments along with the
+   `ignition.platform.id={platform-name}` kernel argument.
+5. Create the grub console.cfg file and apply any platform
+   specific grub console configuration.
+This stage is highly CoreOS specific and subject to change in the
+future if/when we change the way platform specific information is
+defined in our broader efforts to share more defaults with OSBuild.
+"""
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "properties": {
+    "platform": {
+      "description": "The target platform name/ID",
+      "type": "string"
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true
+},
+"mounts": {
+  "type": "array"
+}
+"""
+import json
+import os
+import osbuild.api
+import shutil
+import sys
+
+from osbuild.util import bls
+
+# Constants
+#For console.cfg see https://github.com/coreos/bootupd/pull/620
+#For platforms.json see https://github.com/coreos/coreos-assembler/pull/3709
+GRUB_CONSOLE_CFG = "boot/grub2/console.cfg"
+PLATFORMS_JSON_DEST= "boot/coreos/platforms.json"
+PLATFORMS_JSON_SOURCE = "usr/share/coreos-assembler/platforms.json"
+
+def generate_console_settings_file(console_settings, file_path):
+    settings_content = ""
+    if console_settings is not None:
+        settings_content = "\n".join(console_settings)
+    file_content = f"""
+# Any non-default console settings will be inserted here.
+# CONSOLE-SETTINGS-START
+{settings_content}
+# CONSOLE-SETTINGS-END
+"""
+    with open(file_path, 'w') as file:
+        file.write(file_content)
+
+
+def process_platforms_json(json_file_path, platform):
+    keys = ["grub_commands", "kernel_arguments"]
+    result = {}
+    with open(json_file_path, 'r') as file:
+        data = json.load(file)
+        if platform in data:
+            for key in keys:
+                if key in data[platform]:
+                    result[key] = data[platform][key]
+
+    return result.get("grub_commands", []),\
+           result.get("kernel_arguments", [])
+
+
+def main(paths, tree, options):
+
+    deployment = options.get("deployment")
+    platform = options.get("platform")
+
+    root = args["paths"]["mounts"]
+    boot_path = os.path.join(root, "boot")
+    platforms_source_path = os.path.join(root, PLATFORMS_JSON_SOURCE.lstrip())
```

Comment:
why the `lstrip()` here and in the lines below? AFAICT we defined the constants above and they do not contain whitespaces? or am I missing something?

Reply:
I think it's an artifact of a previous code state/comment I made.

In an earlier iteration of this code we had something like:

```
PLATFORMS_JSON_SOURCE = "/usr/share/coreos-assembler/platforms.json"
...
...
 platforms_source_path = root + PLATFORMS_JSON_SOURCE
```

but then I suggested we use `os.path.join(root, PLATFORMS_JSON_SOURCE.lstrip())` instead (note I should have suggested `lstrip("/")`.

But in a newer iteration of the code she fixed it so there was no prepending `/`:

```
PLATFORMS_JSON_SOURCE = "usr/share/coreos-assembler/platforms.json"
platforms_source_path = os.path.join(root, PLATFORMS_JSON_SOURCE.lstrip())
```

So yeah, there is no need for the `lstrip()` here. If @ravanelli wants to add the `/` back on to the front of PLATFORMS_JSON_SOURCE she could add `.lstrip("/")` here.

Reply:
The reason why I had to remove the `/` is because `os.path.join` doesn't work when we have to full paths, it doesn't merge two separate paths into one. That's why in the first interaction I used `+`.  I will remove `lstrip` in this case.

Reply:
Aha, thank you for providing the background here, now it all makes sense :) Fwiw, I'm not a fan of python and it's opinions about `/` here, but it is what it is :) Thanks for also simplifying the code!